packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -404,10 +395,11 @@ function objectToMapTree(data: InsertableContent, schema: TreeNodeSchema): Exclu
 
 	const fields = new Map<FieldKey, ExclusiveMapTree[]>();
 
-	// Loop through field keys without data, and assign value from its default provider.
+	// Loop through field keys without data.
+	// This does NOT apply defaults.
 	for (const [key, fieldInfo] of schema.flexKeyMap) {
-		const value = (data as Record<string, InsertableContent>)[key as string];
```

Comment:
accessing the value before checking if it has the own property is not great. If we want to support non-own properties (such as inherited getters) we could, but since this doesn't, it probably should run a getter (possible throwing in some cases) then throw the result away. Its much better to only do the property access after checking its an own property.

Also setFieldValue already has handlering logic for undefined, so allowing them to slip through to it is fine.